src/VisualStudio/Core/Def/Implementation/ProjectSystem/RunningDocumentTableEventTracker.cs

```diff
@@ -110,7 +110,14 @@ public int OnAfterAttributeChangeEx(uint docCookie, uint grfAttribs, IVsHierarch
 
         public int OnBeforeDocumentWindowShow(uint docCookie, int fFirstShow, IVsWindowFrame pFrame)
         {
-            return VSConstants.E_NOTIMPL;
+            // Doc data reloaded is not triggered for the underlying aspx.cs file when changes are made to the aspx file, so catch it here.
+            if (fFirstShow != 0 && _runningDocumentTable.IsDocumentInitialized(docCookie) && TryGetMoniker(docCookie, out var moniker) && TryGetBuffer(docCookie, out var buffer))
+            {
+                _runningDocumentTable.GetDocumentHierarchyItem(docCookie, out var hierarchy, out _);
+                _listener.OnOpenDocument(moniker, buffer, hierarchy);
```

Comment:
What happens in the regular .cs case? Is it possible to get both this calling OnOpenDocument and the reload one *also* doing it? I think the regular workspace one won't care but not sure if the miscellaneous files workspace will get tripped up.

Reply:
Yes - both events will get triggered in that case.

Neither workspace should care about getting multiple events - both already could get multiple open events since the VS workspace tracked initialized doc and before open, while the misc workspace tracked initialized doc and doc data reloaded.

This misc workspace has this check which keeps stuff safe - https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs#L208